pkgs/native_assets_builder/lib/src/dependencies_hash_file/dependencies_hash_file.dart

```diff
@@ -135,27 +166,43 @@ class DependenciesHashFile {
 class FileSystemHashes {
   FileSystemHashes({
     List<FilesystemEntityHash>? files,
-  }) : files = files ?? [];
+    List<EnvironmentVariableHash>? environment,
```

Comment:
As I mentioned multiple times before, I think doing `content_hashes` is a bad idea because we can't give good logging/error messages. (Also whatever else content we come up with could collide with the env keys. So we'd want `Map<String, Map<String, String>>`, but that goes against KISS.)

I tried to keep it close to `lib/src/build_system/file_store.dart` from flutter tools. I believe the storage format is separated as a class from the public API class to follow single responsibility principle. (Which is probably also the reason the entries have their own class, but yeah it does create the situation with double keys. Note that string keys don't help. File paths can be the same thing even if the string is not equal. Casing on Windows, and non-normalized paths everywhere.)